### ISSUE TITLE
Need advice : Translating Pal's name and attributes into native language

### ISSUE THREAD

**Issue by YupGoolg** [tylercamp/palcalc#29]
I translating Pal's name and attributes into native language.

I modified the 'names' of the two files scraped-pals.json and scraped-traits.json and created a DB.json file with GenDB.
When you debug your program, the names and attributes are translated and displayed correctly.
However, two problems arise:
The friend icon will change to human, and setting the attribute will show no results.
Change the pal icon file name, a debug error will occur.

What file links a friend's name to a friend's icon?
What file is calling the attributes?

And I want to list names in order of friend number. What should I do? (I'm thinking about putting a number in front of the Pal name.)

**Comment by tylercamp** [comment]
> What file links a friend's name to a friend's icon?

The display name of the pals are used to get the icon, I'll change it from `Name` to `InternalName` later so you don't have this issue.

You can use `PalCalc.UI/Resources/PalIconOverride.json` to manually change the icon for each pal

> And I want to list names in order of friend number. What should I do? (I'm thinking about putting a number in front of the Pal name.)

Are you changing an existing list? Or adding a new one? Do you mean the list of target pals on the left?

**Comment by YupGoolg** [comment]
The name change and icon application have been completed.

The list of target pals on the right in the attached photo is in ABCD order.
I would like to list these in order of pal number.

And if you look at the properties in the middle pal photo, the changed property names are not applied properly. Where can I manually modify Pal characteristics?

All names in the scraped-traits.json file have been changed.
![스크린샷 2024-07-10 072056](https://github.com/tylercamp/palcalc/assets/96374741/a1beafff-0147-4496-a439-d9778d2f2c21)

**Comment by tylercamp** [comment]
> The list of target pals on the right in the attached photo is in ABCD order.
> I would like to list these in order of pal number.

That's handled by the `PalTargetView`, whose main logic is in `PalTargetViewModel`. If you look at `PalTargetView.xaml`, the dropdown has `ItemsSource={Binding AvailablePals}`, and you'll see the `OrderBy(p => p.Name)` below it. That's what you should change

> And if you look at the trait in the middle pal photo, the changed trait names are not applied properly. Where can I manually modify Pal trait?

You may need to re-run `PalDB.BuildDBProgram` and update the DB version (`var db = PalDB.MakeEmptyUnsafe("v11");`) for changes to the PalDB to apply

**Comment by YupGoolg** [comment]
Thank you for continuing to respond.

I modified the translation of the name in scraped-traits.json.
![스크린샷 2024-07-11 014852](https://github.com/tylercamp/palcalc/assets/96374741/ff177c5d-de86-4791-a39b-dde6166f23ac)

It is displayed normally in the list on the right, but when you select a traits and run it, the pal is not recognized.
![스크린샷 2024-07-11 014355](https://github.com/tylercamp/palcalc/assets/96374741/ba5446d6-f2b0-45db-be82-f5faa0e61c45)

When breeding a pal randomly without selecting a trait, the name of the trait whose name has been changed is displayed in the pal's traits, but it is displayed as unrecognized (ex.lucky).
![스크린샷 2024-07-11 014717](https://github.com/tylercamp/palcalc/assets/96374741/0c1aaf1e-fc16-4b0b-9a28-4c07ee005b78)

We need file information so we can change the code that calls scraped-traits.json.

And I'm wondering what I need to update in (var db = PalDB.MakeEmptyUnsafe("v11") in PalDB.BuildDBProgram

**Comment by tylercamp** [comment]
The trait is shown as unrecognized since the name stored in the cached save still uses the old english name, you'll need Pal Calc to regenerate that file. You can do this manually by deleting `PalCalc.UI\bin\Debug\net8.0-windows10.0.17763.0\cache`.

For the code in BuildDBProgram, change `v11` to `v12` (or anything else), and Pal Calc will detect that the cache file is out of date and regenerate it. This is the preferred way to regenerate the cache file, where Pal Calc does it automatically (since the version `v11` was changed), but you can also do it manually by deleting the folder I mentioned.

**Comment by YupGoolg** [comment]
Excellence! Your app is really great.
Your app is becoming widely known in Korea.
My material always cites your original source.
Thank you for creating such a great program.

Additionally, like traits, Pal's skills can also be transferred.
Are you thinking of implementing Pal's skills?

If necessary, I would also like to translate the UI name in the view.
If you simply change the name, an error will still occur.

If necessary, we can return the source code once all Korean translations are completed.

**Comment by tylercamp** [comment]
Thank you, I'm happy to hear it!

> Additionally, like traits, Pal's skills can also be transferred.
> Are you thinking of implementing Pal's skills?

There's a similar question for stat inheritance, the issue is that I need to know the inheritance probabilities for active skills so I can accurately include these in the breeding calculation.

> If necessary, I would also like to translate the UI name in the view.
> If you simply change the name, an error will still occur.

Which name do you mean?

> If necessary, we can return the source code once all Korean translations are completed.

Right now I'm focusing on #27 but after that I plan to add translation handling in the app to make your work easier, I'll accept translations once I make that change

**Comment by YupGoolg** [comment]
no. There was an error in the panel title, but it has been corrected. thank you
I look forward to your development^^

**Comment by YupGoolg** [comment]
![스크린샷 2024-07-11 145345](https://github.com/tylercamp/palcalc/assets/96374741/e5255aba-da61-4e01-b1ff-3626e7619ba8)
If there are new traits added after 3.2.0, an error occurs when loading the save folder.
Any advice?

**Comment by YupGoolg** [comment]
![스크린샷 2024-07-12 003303](https://github.com/tylercamp/palcalc/assets/96374741/6f563275-9539-43d4-81a9-db044e9aacf6)

Translation completed. Everything is perfect.

What if the gender was an icon like ♂♀ instead of text?
If you are of any gender, two icons are displayed at the same time,
If it is the opposite of one parent, it is expressed like this: ♂↔♀.

**Comment by tylercamp** [comment]
I was hoping to do the gender icons at some point, I just couldn't find any good images with a transparent background. I didn't notice you can write those icons as text, so I'll consider adding that in soon

**Comment by tylercamp** [comment]
@YupGoolg I'm in the process of adding a "Language" dropdown and need to add an entry for each language, is this correct for "Korean"? I got this from ChatGPT: `한국어 (한국말)`
